src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -388,6 +389,7 @@ private async Task WaitForCompletionAsync(IItemLoader<PackageItemViewModel> curr
             while (currentLoader.State.LoadingStatus == LoadingStatus.Loading)
             {
                 token.ThrowIfCancellationRequested();
+                await Task.Delay(PollingDelay, token);
```

Comment:
Why not change the while loop and update only if `currentLoader.State.LoadingStatus` has changed ?

Reply:
... and remove the delay.

Reply:
Seems we'd need a design change of either,
1. Stop aggregating all sources, and instead monitoring them individually for state changes.
2. or, make `HandleItemLoaderStateChange` smarter such that a comparison for the State object is checked for deltas prior to actually blocking the thread and updating UI & bound properties.

Reply:
To elaborate slightly, `RefreshSearchAsync` polls all the sources and creates a dictionary _every_ iteration of the loop. A more MVVM approach would be to have a model where sources populate a collection, which is bound in the UI, and once the model is done, it sets a flag accordingly, so the UI knows it's done. Once all are done, we can infer loading is complete without needing to block and ask all sources repeatedly.